src/vs/workbench/contrib/comments/browser/commentThreadZoneWidget.ts

```diff
@@ -201,7 +201,17 @@ export class ReviewZoneWidget extends ZoneWidget implements ICommentThreadWidget
 						let newPosition = this.getPosition();
 
 						if (newPosition) {
-							this.commentService.updateCommentThreadTemplate(this.owner, this._commentThread.commentThreadHandle, new Range(newPosition.lineNumber, 1, newPosition.lineNumber, 1));
+							let range: Range;
+							const originalRange = this._commentThread.range;
+							if (newPosition.lineNumber !== originalRange.endLineNumber) {
+								// The widget could have moved as a result of editor changes.
+								// We need to try to calculate the new, more correct, range for the comment.
+								const distance = newPosition.lineNumber - this._commentThread.range.endLineNumber;
```

Comment:
We used the zone widget position since previously we only track single line and it serves good enough as markers. To accurately track the movement of the comment range, we may want to add a decoration for `commentThread.range` in the editor and reveive the latest range of the decoration on action execution.

Reply:
This makes sense to me. I will leave as is for now though, since the decoration for a range requires some UX decisions that we have yet to make.